### PR TITLE
Setup tsx components

### PR DESCRIPTION
**Pull request by jroebu14**
Resolves WSTEAMA-123

**Overall change:**
Enables TypeScript and TypeScript React components.

Uses Babel as the TypeScript compiler instead of using the compiler from TypeScript. This means we can continue using Babel to provide polyfills and handle the modern/legacy JS bundle differential serving technique. Achieving these things with the TypeScript compiler would require a complete rethink and after some reading around, it seems it might not be possible.

TypeScript type checking is configured in ESLint so that files are statically analysed with VSCode to find problems. Also added is a `type-check` yarn script that is run after every `yarn build` and can also be run independently. This will catch any type errors after a build so even if we somehow miss them while developing locally, CI will fail and alert us to any type errors.

**Code changes:**

- updates webpack config to handle loading of ts and tsx files using `ts-loader`
- updates jest config with `ts-jest` preset for unit testing
- adds `tsconfig.json` which was inspired by this article on best practices https://www.sitepoint.com/react-with-typescript-best-practices/

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project
- [x] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [x] This PR requires manual testing

### CODE REVIEW

File: src/app/pages/StoryPage/StoryPage.jsx [hunk 1]
```diff
@@ -381,6 +382,7 @@ const StoryPage = ({ pageData, mostReadEndpointOverride }) => {
           parentColumns={gridColumns}
         >
           <main role="main">
+            <Text>Hellooooo I am TypeScript!</Text>
```
Comment (RetiredBlob) [inline src/app/pages/StoryPage/StoryPage.jsx]: Does this pass the CI checks?  I thought `className` would need to be passed as it's not marked as optional in `TextProps`.

**Review by RetiredBlob — APPROVED** [review]
Nice

File: src/app/pages/StoryPage/StoryPage.jsx (hunk 1 above)
Comment (jroebu14) [reply src/app/pages/StoryPage/StoryPage.jsx]: I think because this is jsx that it's not picked up here

Comment (jroebu14) [reply src/app/pages/StoryPage/StoryPage.jsx]: Yeh it is. When I use that component in a tsx file then I get the errors

Screenshot 2022-07-28 at 13 11 37

Comment (RetiredBlob) [reply src/app/pages/StoryPage/StoryPage.jsx]: Aah of course - cheers 👍

**Comment by pvaliani** [comment]
Fancy doing a swarm review for this? Might be good to talk in more detail about how easy it's been to configure. Still reviewing independently. Good job!

**Review by pvaliani — APPROVED** [review]
Looks good! :)

**Comment by amoore108** [comment]
Looks great! One thing that may need added to the `tsconfig.compilerOptions` is the directory aliasing found in `dirAlias`. I don't think the `.tsx` files will be able to find or import from those directories without something like:
```javascript
    "baseUrl": ".",
    "paths": {
      "#components/*": ["src/app/components/*"],
      "#containers/*": ["src/app/containers/*"],
      "#psammead/*": ["src/app/legacy/psammead/*"],
       ...etc
    }
```

**Comment by jroebu14** [comment]
@amoore108 one of the standards we talked about but I forgot to write anything about is not having directory aliases. We can chat about it another time though

**Comment by amoore108** [comment]
> @amoore108 one of the standards we talked about but I forgot to write anything about is not having directory aliases. We can chat about it another time though

Ah yea thats right, I remember that being brought up. TS path aliasing makes the `cmd + click` on a path work a lot more reliably than the current setup, but it doesn't get around the obfuscation problem of course.

Just so long as people know that the current way we import files/components won't work in `.ts / .tsx` files then its grand 👍🏼

**Review by amoore108 — APPROVED** [review]

**Comment by jroebu14** [comment]
@amoore108 cool. I don't think we'll be importing anything those directories anyway so it shouldn't be a problem

**Comment by paruchurisilpa** [comment]
Have performed some regression using few URL's from below list:

https://github.com/bbc/simorgh/blob/latest/docs/Simorgh-Release-Info.md

LGTM..

**Pull request merged**